gson/src/main/java/com/google/gson/stream/JsonReader.java

```diff
@@ -1563,25 +1563,30 @@ private IOException syntaxError(String message) throws IOException {
   }
 
   /**
-   * Consumes the non-execute prefix if it exists.
+   * Consumes the non-execute prefix if it exists and return whether eof has
+   * been reached.
    */
-  private void consumeNonExecutePrefix() throws IOException {
+  private boolean consumeNonExecutePrefix() throws IOException {
     // fast forward through the leading whitespace
-    nextNonWhitespace(true);
-    pos--;
+    int c = nextNonWhitespace(false);
+    if (c == -1) {
+      return true;
+    } else {
+      pos--;
+    }
 
     int p = pos;
     if (p + 5 > limit && !fillBuffer(5)) {
-      return;
+      return false;
     }
 
     char[] buf = buffer;
-    if(buf[p] != ')' || buf[p + 1] != ']' || buf[p + 2] != '}' || buf[p + 3] != '\'' || buf[p + 4] != '\n') {
-      return; // not a security token!
+    if(! (buf[p] != ')' || buf[p + 1] != ']' || buf[p + 2] != '}' ||
+          buf[p + 3] != '\'' || buf[p + 4] != '\n')) {
```

Comment:
Might be good to remove the doubled negation:
```suggestion
    if (buf[p] == ')' && buf[p + 1] == ']' && buf[p + 2] == '}' && buf[p + 3] == '\'' && buf[p + 4] == '\n') {
```